src/irr/asset/CGLTFLoader.cpp

```diff
@@ -29,39 +37,53 @@ namespace irr
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view fragment = ""; // TODO - include/builtin/...
-			}
-
-			namespace NoUVAndColor
-			{
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view vertex = ""; // TODO - include/builtin/...
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view fragment = ""; // TODO - include/builtin/...
-			}
-		}
-
 		/*
 			Each glTF asset must have an asset property. 
 			In fact, it's the only required top-level property
 			for JSON to be a valid glTF.
 		*/
 
+		CGLTFLoader::CGLTFLoader(asset::IAssetManager* _m_assetMgr) 
+			: assetManager(_m_assetMgr)
+		{
+			auto registerShader = [&](auto constexprStringType, ICPUSpecializedShader::E_SHADER_STAGE stage) -> void
+			{
+				auto shaderData = assetManager->getFileSystem()->loadBuiltinData<decltype(constexprStringType)>();
+				auto unspecializedShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(shaderData), asset::ICPUShader::buffer_contains_glsl);
+
+				ICPUSpecializedShader::SInfo specInfo({}, nullptr, "main", stage, stage != ICPUSpecializedShader::ESS_VERTEX ? "?IrrlichtBAW glTFLoader FragmentShader?" : "?IrrlichtBAW glTFLoader VertexShader?");
+				auto cpuShader = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(unspecializedShader), std::move(specInfo));
+
+				auto insertShaderIntoCache = [&](const char* path)
+				{
+					asset::SAssetBundle bundle({ cpuShader });
+					assetManager->changeAssetKey(bundle, path);
+					assetManager->insertAssetIntoCache(bundle);
+				};
+
+				insertShaderIntoCache(decltype(constexprStringType)::value);
+			};
+
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(VERT_SHADER_UV_CACHE_KEY) {}, ICPUSpecializedShader::ESS_VERTEX);
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(VERT_SHADER_COLOR_CACHE_KEY) {}, ICPUSpecializedShader::ESS_VERTEX);
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(VERT_SHADER_NO_UV_COLOR_CACHE_KEY) {}, ICPUSpecializedShader::ESS_VERTEX);
+
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(FRAG_SHADER_UV_CACHE_KEY) {}, ICPUSpecializedShader::ESS_FRAGMENT);
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(FRAG_SHADER_COLOR_CACHE_KEY) {}, ICPUSpecializedShader::ESS_FRAGMENT);
+			registerShader(IRR_CORE_UNIQUE_STRING_LITERAL_TYPE(FRAG_SHADER_NO_UV_COLOR_CACHE_KEY) {}, ICPUSpecializedShader::ESS_FRAGMENT);
+		}
+		
 		bool CGLTFLoader::isALoadableFileFormat(io::IReadFile* _file) const
 		{
 			simdjson::dom::parser parser;
 
-			auto jsonglTFFileBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(_file->getSize());
-			simdjson::dom::object tweets = parser.parse(reinterpret_cast<uint8_t*>(jsonglTFFileBuffer->getPointer()), jsonglTFFileBuffer->getSize());
+			auto jsonBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(_file->getSize());
+			{
+				const auto beginPosition = _file->getPos();
+				_file->read(jsonBuffer->getPointer(), jsonBuffer->getSize());
+				_file->seek(beginPosition);
+			}
```

Comment:
reading the whole file just to check if its a candidate for loading is overkill.

Reply:
maybe dont use the json parser here.